CMakeLists.txt

```diff
@@ -4,7 +4,7 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 # cmake-format: on
 
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.23)
```

Comment:
The build in practice already required 3.20 before, because that is when `CMAKE_CXX_STANDARD` 23 was introduced. Bumping it all the way to 3.23 allows us to use `FILE_SET`.